### PR TITLE
Added a Matcha connector

### PR DESCRIPTION
**Pull request by true-eye**
For reference,
- https://github.com/InstaDApp/dsa-connectors/blob/bd1d510a6a411262909504b578f748bf2f92cfdd/contracts/connectors/1inch.sol#L615
- Readme of https://github.com/InstaDApp/dsa-connectors
- https://0x.org/docs/guides/use-0x-api-liquidity-in-your-smart-contracts#overview
- https://github.com/0xProject/0x-api-starter-guide-code

### CODE REVIEW

**Comment by thrilok209** [comment]
Hey @true-eye! I think 0x API uses WETH instead of ETH for swapping. So, we need to manage the conversation between WETH <> ETH.

File: contracts/connectors/matcha.sol [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import {TokenInterface, MemoryInterface, EventInterface} from "../common/interfaces.sol";
+import {Stores} from "../common/stores.sol";
+import {DSMath} from "../common/math.sol";
+
+contract MatchaHelpers is Stores, DSMath {
+    /**
+     * @dev Return Matcha Address
+     */
+    function getMatchaAddress() internal pure returns (address) {
+        return 0x61935CbDd02287B511119DDb11Aeb42F1593b7Ef;
```
Comment (thrilok209) [inline contracts/connectors/matcha.sol]: Hey @true-eye ! Is this the correct address?

When I am trying to get it from here: https://api.0x.org/swap/v1/quote?buyToken=DAI&sellToken=USDC&sellAmount=10000000000000000000000

`to` address is different.

**Comment by Sowmayjain** [comment]
Quick question: @true-eye - do you know or have any previous experiences with thegraph protocol? (used for querying ethereum blockchain).

**Comment by true-eye** [comment]
> Quick question: @true-eye - do you know or have any previous experiences with thegraph protocol? (used for querying ethereal blockchain).

Yes

File: contracts/connectors/matcha.sol (hunk 1 above)
Comment (true-eye) [reply contracts/connectors/matcha.sol]: Oh, it seems you are right. I will update

Comment (true-eye) [reply contracts/connectors/matcha.sol]: I thought no needed to test but I think I need to.

Comment (thrilok209) [reply contracts/connectors/matcha.sol]: Curious to know, how are you thinking to test it?

**Comment by Sowmayjain** [comment]
@true-eye could you also add the test-cases & hardhat integration in the repo?

**Comment by true-eye** [comment]
> @true-eye could you also add the test-cases & hardhat integration in the repo?

yes

**Comment by Sowmayjain** [comment]
Awesome! Looking forward to it @true-eye

File: contracts/connectors/matcha.sol (hunk 1 above)
Comment (true-eye) [reply contracts/connectors/matcha.sol]: I was thinking that I can integrate my contract to 0x-api-starter-guide-code and replace it with SimpleTokenSwap.

Comment (true-eye) [reply contracts/connectors/matcha.sol]: I've updated the Matcha Address and added conversion between ETH <=> WETH

**Pull request closed without merging**